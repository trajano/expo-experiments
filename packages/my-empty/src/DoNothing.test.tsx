import { render, renderHook, screen } from '@testing-library/react-native';
import { Text } from 'react-native';
import {
  DoNothingContext,
  DoNothingProvider,
  useDoNothing,
  WithDoNothing,
} from './DoNothing';
import { FC, PropsWithChildren } from 'react';

const TestComponent: FC = () => {
  const doNothingProps = useDoNothing();

  return <Text testID="props">{JSON.stringify(doNothingProps)}</Text>;
};

describe('DoNothing', () => {
  it('Use provider', async () => {
    render(
      <DoNothingProvider>
        <TestComponent />
      </DoNothingProvider>,
    );
    expect(screen.getByTestId('props').props.children).toBe('{}');
    expect(screen.toJSON()).toMatchSnapshot();
  });

  it('Use Hook', async () => {
    // eslint-disable-next-line @typescript-eslint/no-empty-object-type
    const wrapper = ({ children }: PropsWithChildren<{}>) => (
      <DoNothingContext.Provider value={{}}>
        {children}
      </DoNothingContext.Provider>
    );
    const { result } = renderHook(() => useDoNothing(), { wrapper });
    expect(result.current).toEqual({});
  });

  it('Use HoC', async () => {
    const TestedComponent = WithDoNothing(TestComponent);
    render(<TestedComponent />);
    expect(screen.getByTestId('props').props.children).toBe('{}');
    expect(screen.toJSON()).toMatchSnapshot();
  });
});
