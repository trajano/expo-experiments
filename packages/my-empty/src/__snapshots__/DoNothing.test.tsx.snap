// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DoNothing Use HoC 1`] = `
<Text
  testID="props"
>
  {}
</Text>
`;

exports[`DoNothing Use provider 1`] = `
<Text
  testID="props"
>
  {}
</Text>
`;
